File: 0409-longest-palindrome/0409-longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        dictt = set()
        for char in s:
            if char not in dictt:
                dictt.add(char)
            else:
                dictt.remove(char)
        if len(dictt) != 0:
            return len(s) - len(dictt) + 1
        return len(s)

File: 0409-longest-palindrome/0409-longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        map<char, int> m;
        int oddCount = 0;
        for(int i=0; i<s.length(); i++){
            m[s[i]]++;
            if(m[s[i]]%2 == 1){
                oddCount++;
            }else{
                oddCount--;
            }
        }
        if(oddCount > 1){
            return s.length()-oddCount+1;
        }
        return s.length();
    }
};